yt/data_objects/time_series.py

```diff
@@ -195,7 +195,9 @@ def __getitem__(self, key):
             if isinstance(key.start, float):
                 return self.get_range(key.start, key.stop)
             # This will return a sliced up object!
-            return DatasetSeries(self._pre_outputs[key], self.parallel)
+            return DatasetSeries(self._pre_outputs[key],
+                                 parallel=self.parallel,
+                                 **self.kwargs)
```

Comment:
do we need to do the same thing with `setup_function`?

Reply:
I don't know what it's supposed to do, but I don't think so. Apparently everywhere a `self._setup_function` is called upon a given `ds`, it happens right after said `ds` is initialised and passed the kwargs.

Reply:
You can create a time series from a list of datasets, so you can do something like:

```
ts = DatasetSeries([fake_random_ds(16) for i in range(n_steps)])
```

Like this test: https://github.com/yt-project/yt/blob/master/yt/data_objects/tests/test_particle_trajectories.py#L54

Reply:
I don't think it will work here, because the behaviour I want to test is for datasets that are  instantiated by the `DatasetSeries` container, which only happens if `DatasetSeries._pre_outputs` contains strings-likes, rather that existing datasets (see https://github.com/yt-project/yt/blob/e76f71c23bb274e7fa53bc7a6c40ac8e751bd284/yt/data_objects/time_series.py#L200)

This is turning into a separate issue.

Reply:
Ah that makes sense. In that case just write a test with a real time series dataset from yt-project.org/data.

Or don’t 🤷

Tests aren’t necessary for bugfix PRs.

Reply:
I'd rather do it another time, when I feel more confortable with timeseries as a user :)